### PR TITLE
650-add-plugin-store-list-to-manager

### PR DESCRIPTION
**Pull request by Pierregvx**

### CODE REVIEW

**Comment by Pierregvx** [comment]
run the test with with `go test ./api/test -v`

File: pkg/plugin/manager.go [hunk 1]
```diff
@@ -41,18 +46,39 @@ type Manager struct {
...
 
-	err := manager.RunAll()
 	if err != nil {
-		return manager, fmt.Errorf("while running all plugin: %w", err)
+		return manager, fmt.Errorf("while running all plugins: %w", err)
 	}
 
+	const interval = 20
+	// Start a goroutine to fetch the store every ten minutes
+	go func() {
```
Comment (peterjah) [inline pkg/plugin/manager.go:68]: move this function implementation elsewhere with an appropriate name!

File: pkg/plugin/manager.go [hunk 2]
```diff
@@ -303,10 +328,73 @@ func (m *Manager) Update(correlationID string) error {
...
+	originalDirector := plgn.reverseProxy.Director
+	plgn.reverseProxy.Director = func(req *http.Request) {
+		originalDirector(req)
+		modifyRequest(req)
+	}
+
+	return nil
+}
+
+func findPluginByName(name string, plugins []store.Plugin) *store.Plugin {
```
Comment (peterjah) [inline pkg/plugin/manager.go:370]: move this func in pkg/store

**Review by peterjah — CHANGES_REQUESTED** [review]

File: pkg/store/store.go [hunk 3]
```diff
@@ -30,12 +32,35 @@ type File struct {
...
+func NewStore() (*Store, error) {
+	plgns, err := FetchPluginList()
+	if err != nil {
+		//nolint:exhaust,exhaustruct
+		return &Store{}, fmt.Errorf("while fetching plugin list: %w", err)
+	}
+
+	storeMassaStation := &Store{plugins: plgns}
+
+	interval := 10
```
Comment (peterjah) [inline pkg/store/store.go:58]: define this directly in FetchStorePeriodically, or even in global constant

**Review by peterjah — CHANGES_REQUESTED** [review]

**Review by Thykof — CHANGES_REQUESTED** [review]

**Pull request merged**